File: 8kyu/js/numberToString.js
/**
 *  Convert a Number to a String!
 *  We need a function that can transform a number (integer) into a string.

    What ways of achieving this do you know?

    Examples (input --> output):
    123  --> "123"
    999  --> "999"
    -100 --> "-100"
 */


/*---- Solution ----*/

function numberToString(num) {
  // Return a string of the number here!
  /*
    the first approach i can think about is to typecasting the given input to string
  */
  
  return num.toString();
}

/*---- Sample Tests ----*/
const assert = require('chai').assert;
describe("Tests", () => {
  it("test", () => {
    assert.strictEqual(numberToString(67), '67');
  });
});

File: 8kyu/java/numberToString.java
/**
 *  Convert a Number to a String!
 *  We need a function that can transform a number (integer) into a string.

    What ways of achieving this do you know?

    Examples (input --> output):
    123  --> "123"
    999  --> "999"
    -100 --> "-100"
 */

/* Solution */
class Kata {
  public static String numberToString(int num) {
    
    //for java you have mention Integer.toString() instead of .toString()
    return Integer.toString(num); // Return a string of the number here!
  }
}

/* Sample Tests */

import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class NumberStringExampleTests {
  @Test
  public void tests() {
    assertEquals("67", Kata.numberToString(67));
    assertEquals("123", Kata.numberToString(123));
    assertEquals("999", Kata.numberToString(999));
    assertEquals("0", Kata.numberToString(0));
  }
}

